File: Answer/1288.删除被覆盖区间.java
/*
 * @lc app=leetcode.cn id=1288 lang=java
 *
 * [1288] 删除被覆盖区间
 */

import java.util.Arrays;

// @lc code=start
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> {
            return a[0] != b[0] ? a[0] - b[0] : b[1] - a[1];
        });

        int ans = 0;
        int end = 0;
        for (int[] it : intervals) {
            if (it[1] > end) {
                ans++;
                end = it[1];
            }
        }
        return ans;
    }
}
// @lc code=end

File: Answer/1288.删除被覆盖区间.py
#
# @lc app=leetcode.cn id=1288 lang=python3
#
# [1288] 删除被覆盖区间
#

# @lc code=start
from typing import List


class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))
        ans = 0
        end = 0
        for l, r in intervals:
            if r > end:
                ans += 1
                end = r
        return ans
# @lc code=end
